ncl/receipts/trigonometric_functions.ipynb

```diff
@@ -0,0 +1,275 @@
+{
```

Comment:
<p>Are these diffs real?  If you provide a double to NCL's sin you should return a double as well which I imagine would be closer.  </p><p> </p><p>I think Python's built in float type that Math uses is double precision. </p><p> </p><p>Might be good to have some discussion of this somewhere (if it's not already elsewhere)</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NCAR/geocat-applications/pull/40/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
NCL is using a `typeof(0.5) = float` and Python is using `type(0.5) = float`. Am I missing a NCL feature? When a value is converted to a double, it still returns the same level of precision when using `math`
```
math.asin(np.double(0.5))
0.5235987755982989
```

Reply:
There are some diffs in the definitions depending upon the context.  

In NCL the `float` type is 32 bits and `double` 64 bits, whereas in Python the `float` type is generally a 64-bit floating point number.  

This is also slightly different from numpy which has its own types (e.g. `np.float64`, `np.float32`).  

Generally the 64-bit types are the defaults in Python though, which is different from NCL and something that's come up in the past.

It seems like this is also what's causing the diffs here.

Reply:
I can expand on the reasons behind the differences in this section of the receipt, but should it be rewritten to match? Currently, the `assert` statement is written to match within 6 decimal points
```
assert math.isclose(
        ncl_results[c], results_math[c], rel_tol=1e-06
    )  # within 6 decimal points
```

Reply:
Maybe depends on what we're looking for in the comparison.  If we're not looking for precision level diffs (@anissa111 thoughts on this? -  you probably have a better idea how often this comes up or not), I think it's fine as is with a note somewhere in here about why.

Reply:
My impression is that this is going to come up a lot. I think it would make sense to make a full entry about the precision differences in the future and link that where/when it becomes relevant.